### ISSUE TITLE
Add <q> tag rendering option

### ISSUE THREAD

**Issue by LAfricain** [crosswire/xiphos#1159]
In French, we don't use English straight quotation marks, like the ones used to render the q tag. Often, the q tag designates quotations that are italicized in the original text. In bibletime, italics have been chosen. Is it possible to change this option in the css so as not to have the right quotation marks? If not, would it be possible to set options to avoid this display in French?

**Comment by LAfricain** [comment]
Have a look with freCrampon or FreVuglGlaire. Gn 1,1 you have 3 notes in FreCrampon, and one for FreVulg Glaire.
I particularly like Andbible's method, which displays it in bold gray. Otherwise, I'm interested in css. Could you give me the necessary inputs. Would it be possible to add a graphics way for this kind of tweak?

**Comment by karlkleinpaste** [comment]
Between...
```
mod2imp FreCrampon | grep -A1 Genesis.1:1\$ | grep '<q'
```
...and...
```
diatheke -b FreCrampon -o f -f xhtml -k gen.1.1
```
...I'm honestly not sure what's in question. All the `<q>` in these notes appear to reference just eID and sID, and are self-closing. I see no relation to use of quotation marks.

I don't have an Android, so I don't know what's unique about AndBible here.

I simply don't know what you're asking.

**Comment by LAfricain** [comment]
You can see in the notes to the examples I've given you that there are English quotation marks (") Ex: Dieu, hébr. "Elohim". But these quotation marks are not in use in French. So my idea is either to render q not in quotation marks but, for example, in italics, as is the case in the paper versions of the Bibles concerned.
Basically, just make the q tag not with quotation marks but with something more universal.
Is it any clearer?
About Andbible, I made a mistake and confused it with catchword.

**Comment by karlkleinpaste** [comment]
This is just symptomatic of the fact that I don't really do OSIS. (I was always quite happy with ThML, being "HTML plus goodies," namely `<sync>`, `<scripRef>`, and `<note>`.)

Unfortunately, the rendering by `diatheke` doesn't include the footnote content, but rather `passagestudy.jsp` fake/internal refs that induce retrieval of note content when accessed. So I can't see what the rendering looks like using ordinary command line tools.

But what I think I now understand is that you need a CSS class to wrap `<q>` sID/eID content, identifying both start and end, probably in some sort of `<span>`. You'll have to ask sword-devel whether that's even possible. I don't know, but it's certain that I can't help you with it at this point, and I don't think Xiphos can have that sort of control in the absence of an identifiable CSS control.

I'll leave this open for a while in case there's any further input about it.

**Comment by dmsmith** [comment]
> In French, we don't use English straight quotation marks, like the ones used to render the q tag. Often, the q tag designates quotations that are italicized in the original text. In bibletime, italics have been chosen. Is it possible to change this option in the css so as not to have the right quotation marks? If not, would it be possible to set options to avoid this display in French?

Use the marker attribute of the q tag to give the quotation mark that is wanted. Where you have <q sID="xxx" marker="opening marker"/> and <q eID="xxx" marker="closing marker"/>

If you wish to render the content of the quote in italics, probably use <hi type="Italic">content</hi>. If the content is spanning multiple verses, you'll need to have it per verse. The problem with using hi in this fashion is that it loses semantic meaning. There might be a better way to indicate why the quote is to be highlighted.

**Comment by UnasZole** [comment]
To clarify, in the source documents (generated by my Bible scraper), the <q></q> tag is not in milestoned form : no sID, no eID, it's a standard tag which contains the contents of the quote.
As a general rule, I use milestones only for verse and chapter tags, which act as annotations on top of the "real" text structure.

The conversion of these q tags to milestoned form comes from osis2mod, which uses the opposite strategy (milestoning everything except verses and chapters).
So I understand that this strategy, which is optimised for indexing, is making rendering a lot harder than it should be. (Styling a text between <q></q> tags should be trivial)

My intuition is that, since this conversion is performed by the sword toolchain (osis2mod), the opposite conversion should be performed by the sword toolchain when generating html, so that the html can still be styled according to the wishes of the initial document author.
In other words, sword should automatically "unmilestone" the elements before rendering a verse. (And in the case of an element that started before the verse, add a fake opening tag at the beginning as a best effort approximation - and respectively for an element ending after the verse).

You mentioned the <hi> element, which does raise a question : how is <hi> handled today ?
Is it not milestoned at all by osis2mod ?

**Comment by dmsmith** [comment]
> To clarify, in the source documents (generated by my Bible scraper), the `<q></q>` tag is not in milestoned form : no sID, no eID, it's a standard tag which contains the contents of the quote. As a general rule, I use milestones only for verse and chapter tags, which act as annotations on top of the "real" text structure.

The problem with quotes is that they can span more than one verse or even paragraphs, sections or divisions, what we have referred on sword-devel as BSP (book, section, paragraph). This is recommended for OSIS docs, rather than BCV (book, chapter, verse).

In many languages a quote that spans many paragraphs has a quote mark at the beginning of the quote, at the beginning of each paragraph and at the end of the quote.

The other aspect of many languages other than English, the opening and closing quotes are very different. Beyond ASCII, English typically has different quotes. Also, languages differ by level 1 and level 2 quotes. SWORD, if not supplied with the quote markers will put " for first level and ' for second level. I thought I remember that there was a conf entry where alternative quote marks could be supplied, but I searched for a couple of hours yesterday and today for it without success. SWORD has support for the marker attribute to supply the quotation mark exactly where it is wanted an in what manner.

Without this a quote as a container element spanning multiple would place the quote mark at the beginning and the end of the quote container and not also at the beginning of every paragraph start. It is because of this difficulty that many have resorted to putting the quote marks directly into the text.

> 
> The conversion of these q tags to milestoned form comes from osis2mod, which uses the opposite strategy (milestoning everything except verses and chapters). So I understand that this strategy, which is optimised for indexing, is making rendering a lot harder than it should be. (Styling a text between `<q></q>` tags should be trivial)

The strategy is not optimized for indexing, as it doesn't impact indexing. It is so that search results of verses or passages can be displayed with rich markup. Also, JSword uses it for parallel display where verses are displayed in a table.

The core goal of SWORD is rich biblical research that is verse centric. It is not for only seeing chapter at a time. You'll find Troy reiterating this even recently.

> 
> My intuition is that, since this conversion is performed by the sword toolchain (osis2mod), the opposite conversion should be performed by the sword toolchain when generating html, so that the html can still be styled according to the wishes of the initial document author. In other words, sword should automatically "unmilestone" the elements before rendering a verse. (And in the case of an element that started before the verse, add a fake opening tag at the beginning as a best effort approximation - and respectively for an element ending after the verse).

The transformation is lossy. It might be possible to write a render filter that tries to put an entire chapter back in a nearly original fashion, hopefully comparable.

> 
> You mentioned the `<hi>` element, which does raise a question : how is `<hi>` handled today ? Is it not milestoned at all by osis2mod ?

It is not milestoned. It is expected to start and end within a verse. I don't think that osis2mod handles it spanning verses. It probably could by splitting it into parts, still keeping it as a container.

**Comment by karlkleinpaste** [comment]
This discussion having gone far afield to osis2mod and Sword's rendering, it's manifestly not a Xiphos issue until/unless it comes back to a way for Xiphos to detect a difference. And that is clearly not coming.

**Comment by LAfricain** [comment]
An interesting alternative is to use the "marker'' in the <q> tag like this:
`<q level="1" marker="« " sID=""/>Au commencement<q eID="" level="1" marker=" »" />`
